src/lib/auth.ts

```diff
@@ -40,39 +63,141 @@ export const ROLES: {
 	],
 	// also default-user (see below)
 	'named-user-api-key': [
-		'resin.actor.delete?id eq @__ACTOR_ID',
-		'resin.api_key.read?is_of__actor eq @__ACTOR_ID',
-		'resin.application.all',
-		'resin.device_type.read',
-		'resin.device_type_alias.read',
-		'resin.cpu_architecture.read',
-		'resin.application_config_variable.all',
-		'resin.application_environment_variable.all',
-		'resin.application_tag.all',
-		'resin.application_type.all',
-		'resin.device.all',
-		'resin.device.tunnel-22222',
-		'resin.device_config_variable.all',
-		'resin.device_environment_variable.all',
-		'resin.device_tag.all',
-		'resin.device_service_environment_variable.all',
-		'resin.image.all',
-		'resin.image__is_part_of__release.all',
-		'resin.image_environment_variable.all',
-		'resin.image_install.all',
-		'resin.image_label.all',
-		'resin.organization.read',
-		'resin.organization_membership.read',
-		'resin.release.all',
-		'resin.release_tag.all',
-		'resin.service.all',
-		'resin.service_environment_variable.all',
-		'resin.service_install.all',
-		"resin.service_instance.read?service_type eq 'vpn'",
-		'resin.service_label.all',
-		'resin.user.read',
-		`resin.user__has__public_key.all?${matchesUser}`,
-		'resin.release_asset.all',
+		// 'resin.actor.delete?id eq @__ACTOR_ID',
+		resinAuth({
+			resource: 'actor',
+			access: 'delete',
+			options: {
+				$filter: {
+					id: {
+						'@': '__ACTOR_ID',
```

Comment:
I would definitely separate this out as a reusable var, eg `actorId` or something as it would help a bit with deduplication and since it is a constant that has to match precisely there's no sense in retyping it each time